raiden-dapp/src/views/RaidenAccount.vue

```diff
@@ -0,0 +1,225 @@
+    <v-form
+      v-if="!loading && !error"
+      ref="form"
+      v-model="valid"
+      class="raiden-account"
+      @submit.prevent="transfer"
+    >
+      <v-row justify="center" align-content="center">
+        <v-col cols="3" class="raiden-account__column">
+          <span v-if="isFromMainToRaidenAccount">
+            {{ $t('general.main-account') }}
+          </span>
+          <span v-else>
+            {{ $t('general.raiden-account') }}
+          </span>
+        </v-col>
+        <v-col cols="3" class="raiden-account__column">
+          <v-btn icon x-large @click="toggleDirection">
+            <v-icon x-large color="primary">mdi-autorenew</v-icon>
+          </v-btn>
+        </v-col>
+        <v-col cols="3" class="raiden-account__column">
+          <span v-if="isFromMainToRaidenAccount">
+            {{ $t('general.raiden-account') }}
+          </span>
+          <span v-else>
+            {{ $t('general.main-account') }}
+          </span>
+        </v-col>
+      </v-row>
+      <v-row justify="center">
+        <v-col cols="10">
+          <amount-input
+            v-model="amount"
+            :token="token"
+            :max="maximumAmount"
+            limit
+          />
+        </v-col>
+      </v-row>
+      <v-row justify="center">
+        <v-col cols="10">
+          <action-button
+            :enabled="valid"
+            :text="$t('general.buttons.transfer')"
+          />
+        </v-col>
+      </v-row>
+    </v-form>
+    <div v-else-if="error" class="raiden-account__progress-wrapper">
+      <error-message :error="error" />
+    </div>
+    <div v-else class="raiden-account__progress-wrapper">
+      <v-progress-circular
+        :size="125"
+        :width="4"
+        class="address-input__prepend"
+        indeterminate
+        color="primary"
+      />
+      <p class="raiden-account__progress-hint">
+        {{ $t('raiden-account.in-progress') }}
+      </p>
+    </div>
+  </v-row>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator';
+import { mapGetters, mapState } from 'vuex';
+
+import AmountInput from '@/components/AmountInput.vue';
+import ActionButton from '@/components/ActionButton.vue';
+import ErrorMessage from '@/components/ErrorMessage.vue';
+import { Token } from '../model/types';
+import { bigNumberify, parseEther, BigNumber } from 'ethers/utils';
+
+@Component({
+  components: { AmountInput, ActionButton, ErrorMessage },
+  computed: {
+    ...mapState(['raidenAccountBalance', 'accountBalance']),
+    ...mapGetters(['isConnected'])
+  }
+})
+export default class RaidenAccount extends Vue {
+  isConnected!: boolean;
+  isFromMainToRaidenAccount = true;
+  amount = '0.0';
+  loading = false;
+  valid = false;
+  error: Error | null = null;
+  raidenAccountBalance!: string;
+  accountBalance!: string;
+  token: Token = {
+    address: '0x1',
+    decimals: 18,
+    symbol: 'ETH',
+    name: 'Ether',
+    balance: 0
+  };
```

Comment:
This is to cheat the AmountInput component into working with native token. Happy to change it if there's a better way

Reply:
I have no isseus with that, do you think it would make sense to actually integrate it in the amount input instead and make it pre-filled with some flag? like `native`?

Reply:
Good idea, I have created #1432 for that

Reply:
Awesome, thank you :)